service/src/ocall_bridge/ffi/get_update_info.rs

```diff
@@ -16,46 +16,41 @@
 
 */
 
-use crate::ocall_bridge::bridge_api::{Bridge, RemoteAttestationBridge};
+use crate::ocall_bridge::bridge_api::{Bridge, OCallBridgeResult, RemoteAttestationBridge};
 use log::*;
 use sgx_types::{sgx_platform_info_t, sgx_status_t, sgx_update_info_bit_t};
 use std::sync::Arc;
 
 #[no_mangle]
-pub extern "C" fn ocall_get_update_info(
+pub unsafe extern "C" fn ocall_get_update_info(
 	p_platform_blob: *const sgx_platform_info_t,
 	enclave_trusted: i32,
 	p_update_info: *mut sgx_update_info_bit_t,
 ) -> sgx_status_t {
-	get_update_info(
-		p_platform_blob,
+	if p_platform_blob.is_null() || p_update_info.is_null() {
+		return sgx_status_t::SGX_ERROR_INVALID_PARAMETER
+	}
```

Comment:
In general, I think I like your proposal, and agree with it.

The only thing that I wonder; the teaclave examples don't check for null pointers in their sample code either, so I wonder if the overall construct with rust FFI and .edl files gives guarantees such that these function args can't be null?

Reply:
Hmm, yeah you are right regarding the teaclave example, but I don't see how they could guarantee that the arguments are never null. It can be a valid use-case to pass a null into a function so it cannot be enforced in general.

Reply:
I could imagine that the edger8r does enforce it. As the .edl files are strongly typed, the edger8r might enforce that a pointer to something does, in fact, always point to a valid object of said type. I am not sure about this though, but I found this very interesting reference to the [edger8r](https://github.com/openenclave/openenclave/blob/master/docs/GettingStartedDocs/Edger8rGettingStarted.md#getting-started-with-the-open-enclave-edger8r).

Reply:
There is only one place in the code samples where they check for null pointer on the `v2.0.0-preview` branch:
https://github.com/apache/incubator-teaclave-sgx-sdk/blob/v2.0.0-preview/samplecode/httpreq/enclave/src/lib.rs#L33:L38

(There are some checks on the current master branch as well)

---
I did not find anything related to explicit null checking in https://download.01.org/intel-sgx/latest/linux-latest/docs/Intel_SGX_Developer_Reference_Linux_2.18_Open_Source.pdf

I think it is worth to asking it on openenclave though just to have a confirmation.